streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -88,57 +130,55 @@ public ProcessorStateManager(final TaskId taskId,
                                  final Map<String, String> storeToChangelogTopic,
                                  final ChangelogReader changelogReader,
                                  final LogContext logContext) throws ProcessorStateException {
+        this.logPrefix = format("task [%s] ", taskId);
         this.log = logContext.logger(ProcessorStateManager.class);
+
         this.taskId = taskId;
+        this.sourcePartitions = sources;
         this.changelogReader = changelogReader;
-        this.logPrefix = format("task [%s] ", taskId);
-
-        partitionForTopic = new HashMap<>();
-        for (final TopicPartition source : sources) {
-            partitionForTopic.put(source.topic(), source);
-        }
-        offsetLimits = new HashMap<>();
-        standbyRestoredOffsets = new HashMap<>();
-        this.isStandby = isStandby;
-        restoreCallbacks = isStandby ? new HashMap<>() : null;
-        recordConverters = isStandby ? new HashMap<>() : null;
-        this.storeToChangelogTopic = new HashMap<>(storeToChangelogTopic);
+        this.storeToChangelogTopic = storeToChangelogTopic;
 
-        baseDir = stateDirectory.directoryForTask(taskId);
-        checkpointFile = new OffsetCheckpoint(new File(baseDir, CHECKPOINT_FILE_NAME));
+        this.baseDir = stateDirectory.directoryForTask(taskId);
+        this.checkpointFile = new OffsetCheckpoint(new File(baseDir, CHECKPOINT_FILE_NAME));
 
-        try {
-            // load the checkpointed offsets and then delete the checkpoint file
-            initialLoadedCheckpoints = checkpointFile.read();
-
-            checkpointFile.delete();
-            checkpointFile = null;
-            log.trace("Checkpointable offsets read from checkpoint: {}", initialLoadedCheckpoints);
-        } catch (final IOException e) {
-            throw new ProcessorStateException(format("%sError loading and deleting checkpoint file when creating the state manager",
-                logPrefix), e);
-        }
+        this.taskType = isStandby ? TaskType.STANDBY : TaskType.ACTIVE;
 
         log.debug("Created state store manager for task {}", taskId);
     }
 
-    void clearCheckpoints() throws ProcessorStateException {
+    public void loadCheckpoint() {
```

Comment:
This may be called in different classes in John's PR so I leave it public. After we've done that part and realize all classes are still in a single package I will do it.